src/clue.js

```diff
@@ -1,14 +1,176 @@
+    color: "purple"
+    },
+    {
+    firstName: "Kasandra",
+    lastName: "Scarlet",
+    occupation: "Actor",
+    age: 31,
+    description: "She is an A-list movie star with a dark past",
+    image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+    color: "red"
+    },
+    {
+    firstName: "Eleanor",
+    lastName: "Peacock",
+    occupation: "Socialité",
+    age: 36,
+    description: "She is from a wealthy family and uses her status and money to earn popularity",
+    image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+    color: "blue"
+    },
+    {
+    firstName: "Jack",
+    lastName: "Mustard",
+    occupation: "Retired Football player",
+    age: 62,
+    description: "He is a former football player who tries to get by on his former glory",
+    image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+    color: "yellow"
+    }
+];
 
 // Rooms Collection
-const roomsArray = [];
+const roomsArray = [
+    {
+        name: 'Dining Room'
+    },
+    {
+        name: 'Conservatory',
+    },
+    {
+        name: "Kitchen",
+    },
+    {
+        name: "Study",
+    },
+    {
+        name: "Library"
+    },
+    {
+        name: "BilliardRoom"
+    },
+    {
+        name: "Lounge"
+    },
+    {
+        name: "Ballroom"
+    },
+    {
+        name: "Hall"
+    },
+    {
+        name: "Spa"
+    },
+    {
+        name: "LivingRoom"
+    },
+    {
+        name: "Observatory"
+    },
+    {
+        name: "Theater"
+    },
+    {
+        name: "GuestHouse"
+    },
+    {
+        name: "Patio"
+    }
+];
 
 // Weapons Collection
-const weaponsArray = [];
+const weaponsArray = [
+    {
+        name: "rope", 
+        weight: 10
+    },
+    {
+        name: "knife",
+        weight: 8
+    },
+    {
+        name: "candlestick",
+        weight: 2
+    },
+    {
+        name: "dumbbell",
+        weight: 30
+    },
+    {
+        name: "poison",
+        weight: 2
+    },
+    {
+        name: "axe",
+        weight: 15
+    },
+    {
+        name: "bat",
+        weight: 13
+    },
+    {
+        name: "trophy",
+         weight: 25
+    },
+    {
+        name: "pistol",
+        weight: 20
+    }
+];
 
 // ITERATION 2
 
-// ITERATION 3
+
+function selectRandom (array) { 
+if (suspectsArray.length === 0 || roomsArray.length === 0 || weaponsArray.length === 0) {
+    return undefined
+} else {
+    return array[Math.floor(Math.random() * array.length)]
+}
+}
+
+function pickMystery () {
+    const name = (selectRandom(suspectsArray).lastName);
+    const weapon = (selectRandom(weaponsArray).name);
+    const room = (selectRandom(roomsArray).name);
+
+    let mistery = {name, weapon, room}
+    return mistery
+}
+
+function revealMystery () {
+
+
+    return `${mistery.name} killed Mr. Boddy using the ${mistery.weapons} in the ${mistery.room}!"`
+}
```

Comment:
```suggestion
function revealMystery (mistery) {
    return `${mistery.suspect.firstName} ${mistery.suspect.lastName} killed Mr. Boddy using the ${mistery.weapon.name} in the ${mistery.room.name}!"`
}
```
Recibes el mistery como parametro y de ahí extraes lo que necesitas para el string.